src/Windows/Editor/Impl/Text/EditorBufferSnapshot.cs

```diff
@@ -7,21 +7,24 @@
 
 namespace Microsoft.Languages.Editor.Text {
     [DebuggerDisplay("{" + nameof(GetText) + "()}")]
-    public sealed class EditorBufferSnapshot: TextProvider, IEditorBufferSnapshot {
+    public sealed class EditorBufferSnapshot : TextProvider, IEditorBufferSnapshot {
         private readonly ITextSnapshot _snapshot;
 
         [DebuggerStepThrough]
-        public EditorBufferSnapshot(ITextSnapshot snapshot): base(snapshot) {
+        public EditorBufferSnapshot(ITextSnapshot snapshot) : base(snapshot) {
             _snapshot = snapshot;
         }
 
-        public T As<T>() where T:class => _snapshot as T;
-        public IEditorBuffer EditorBuffer => _snapshot.TextBuffer.ToEditorBuffer();
+        public T As<T>() where T : class => _snapshot as T;
+
+        public IEditorBuffer EditorBuffer
+            => _snapshot.TextBuffer.ToEditorBuffer() ?? new EditorBuffer(_snapshot.TextBuffer);
```

Comment:
Generally it is not needed. I guess (from your call stack) may happen that the buffer was never created via the editor. When `EditorBuffer` is created it auto-attached to the text buffer and next `ToEditorBuffer` should use it.